src/componentsNext/NamespaceCreate/index.tsx

```diff
@@ -61,14 +102,32 @@ const NamespaceCreate = ({ close }: { close: () => void }) => {
     },
   });
 
-  const onSubmit: SubmitHandler<FormInput> = ({ name }) => {
-    createNamespace({ name });
+  const onSubmit: SubmitHandler<FormInput> = ({
+    name,
+    ref,
+    url,
+    passphrase,
+    publicKey,
+    privateKey,
+  }) => {
+    createNamespace({
+      name,
+      mirror: { ref, url, passphrase, publicKey, privateKey },
+    });
   };
 
   // you can not submit if the form has not changed or if there are any errors and
   // you have already submitted the form (errors will first show up after submit)
   const disableSubmit = !isDirty || (isSubmitted && !isValid);
 
+  // if the form has errors, we need to re-validate when isMirror or authType
+  // has been changed, after useForm has updated the resolver.
+  useEffect(() => {
+    if (isSubmitted && !isValid) {
+      trigger();
+    }
+  }, [isMirror, authType, isSubmitted, isValid, trigger]);
+
```

Comment:
I think you have already suspected this last week :D I'm not a fan of this effect, but I totally get why you needed it, and I just tried so solve this problem by implicitly calling the trigger function when the authType changes, but this obviously does not work because in that function scope the value of `authType` has not changed yet and the `trigger()` validates against the "old" schema.

I think this is good for now and a searching for a cleaner solution should not block us at the moment.